### ISSUE TITLE
Possible WebUSB version of Scrcpy? 

### ISSUE THREAD

**Issue by KennyPeachKing** [Genymobile/scrcpy#1241]
Hello! Right now, you can only use Scrcpy using a computer, using a program you have to download. But is it possible to create a WebUSB version of Scrcpy that you go to a certain web page, select your device, and allow you to control and see your phone in the browser? I know there is already an [ADB implementation of WebUSB](https://webadb.github.io). This would be quite useful, especially on Chromebooks that are school owned or devices that can't download scrcpy.

**Comment by LBensman** [comment]
@KennyPeachKing, is that for personal or business use?  The reason I ask is that there is a commercial product that does something very similar (disclaimer: I work for company that develops such product), and it's "industrial strength" that wouldn't be suitable for personal use.

**Comment by KennyPeachKing** [comment]
I think for my purposes, it would be for personal use. Maybe open-source?

**Comment by BigDogBoy** [comment]
@LBensman 

> .................there is a commercial product that does something very similar................... <, which would be....?

**Comment by BigDogBoy** [comment]
> @KennyPeachKing, .........there is a commercial product that does something very similar..........

and that would be???

**Comment by jfmherokiller** [comment]
just glancing at srccpy and it seems the two pathways to make this work would be either a reimplmentation for webusb or seeing if compilation via emscripten will work

**Comment by KennyPeachKing** [comment]
Emscripten seems interesting, but I am still wondering if all of the C APIs are compatible with it.

**Comment by LBensman** [comment]
> > @KennyPeachKing, .........there is a commercial product that does something very similar..........
> 
> and that would be???

Oh, it's like an enterprise-level product that is meant for mobile development and quality assurance teams that is used to host a bunch of mobile devices remotely for the teams to run and test their software.  Screen visualization and device control is a part of it, among other stuff it does.  That's why I was asking if that's for personal or work use - it's definitely meant for commercial enterprises and would not be suitable for personal use (and with enterprise level price tag 🤣).

**Comment by LBensman** [comment]
I didn't know you can now access USB bus via browser - learn something new every day.  Having ADB protocol implemented would help quite a bit (though a subset for scrcpy's needs would probably not be that big of a deal to implement with raw USB - not sure, haven't looked at that closely enough to say more definitively).  Since scrcpy server sends H.264 stream, it should be possible to read it and feed it directly to `<video>` element on the page, so my initial instinct says that it's possible.

I am wondering though about realistic use case for it.  Trusting random site out in the cloud to connect my device for it to only send me back my screen seems like would run up against trust issues, given that at USB level you can probably do quite a bit of nasty stuff - at minimum, you'd be able to run any code with shell-level permissions, much like what scrcpy would do to run its server (that's what stopped me from trying out that WebADB site).  And if school owned devices are configured from running things like scrcpy, I can only imagine that they would be also configured to reject WebUSB API access.

It may be a fun project to try to do, but I'd be curious to understand realistic use-case for it.

**Comment by KennyPeachKing** [comment]
https://github.com/NetrisTV/ws-scrcpy/

This uses a live H.264 stream from a phone using a server hosted on a computer. It uses scrcpy's wireless screen feature.

**Comment by yume-chan** [comment]
Yes it's possible. I made one here: https://github.com/yume-chan/ya-webadb
